src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -733,10 +775,19 @@ void Hole::updateDiameterParam()
             pitch = ThreadPitch.getValue();
         }
 
-        /* Use thread tap diameter, normal D - pitch */
-        diameter = diameter - pitch;
+        // use normed diameters if possible
+        std::string threadType = ThreadType.getValueAsString();
+        if (threadType == "ISOMetricProfile" || threadType == "UNC"
+            || threadType == "UNF" || threadType == "UNEF") {
+            diameter = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].CoreHole;

```

Comment:
@donovaly sorry for necromancing this PR but isn't this a bit dangerous since `TreadSize.getValue()` can return -1? (I find no checks against it). Here's a link to [PropertyEnumeration::getValue()](https://github.com/FreeCAD/FreeCAD/blob/master/src/App/PropertyStandard.cpp#L366) and a link to [Enumeration::getInt()](https://github.com/FreeCAD/FreeCAD/blob/master/src/App/Enumeration.cpp#L255) as reference.

Reply:
Line 750 guards against that, I believe.

Reply:
> Line 750 guards against that, I believe.

Yes, exactly.

> @donovaly sorry for necromancing this PR 

When you think there is a bug is already merged code, please open a new form thread and make a PR with a proposed fix against current master. This way it gets the necessary attraction to review and judge.